File: CPP/2357MakeArrayZeroBySubtractingEqualAmounts.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

int minimumOperations(vector<int>& nums) {
    int nonZeroCount = 0;
    unordered_set<int> seen;

    for (int i = 0; i < nums.size(); i++) {
        if ((nums[i] != 0) && (seen.count(nums[i]) == 0)) {
            nonZeroCount++;
            seen.insert(nums[i]);
        }
    }

    return nonZeroCount;
}

int main() {
    vector<int> nums = {1, 5, 0, 3, 5};
    int result = minimumOperations(nums);

    cout << "\nThe minimum number of operations required to make the array zero is ";
    cout << result << endl;
    return 0;
}

File: Python/2357MakeArrayZeroBySubtractingEqualAmounts.py
def minimumOperations(nums : list[int]) -> int:
    seen = set()
    nonZeroCount = 0

    for i in range(len(nums)):
        if (nums[i] != 0) and (nums[i] not in seen):
            nonZeroCount += 1
            seen.add(nums[i])

    return nonZeroCount

nums = [1, 5, 0, 3, 5]
result = minimumOperations(nums)

print(f"The minimum number of operations required to make the array zero is: {result}")
